pkg/controllers/provisioning/scheduling/nodeset.go

```diff
@@ -76,3 +102,583 @@ func (s *NodeSet) getDaemons(ctx context.Context, constraints *v1alpha5.Constrai
 func (s *NodeSet) Add(node *Node) {
 	s.nodes = append(s.nodes, node)
 }
+
+func (s *NodeSet) SchedulePod(ctx context.Context, pod *v1.Pod) error {
+	// copy the pod as this method modifies the pod's node selectors to force it to meet topology constraints
+	pod = pod.DeepCopy()
+
+	var possibleNodes []*Node
+	for _, node := range s.nodes {
+		if err := node.Compatible(ctx, pod); err == nil {
+			possibleNodes = append(possibleNodes, node)
+		}
+	}
+
+	tightened := s.constraints.Requirements.Add(v1alpha5.NewPodRequirements(pod).Requirements...)
```

Comment:
Isn't this duplicating requirements math with `node.Compatible`?

Reply:
I don't think so, this set of tightened requirements gets modified by the topology decisions we make and is used to inform future topology decisions.